connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -137,8 +138,21 @@ public void initialize(TaskConfig taskConfig) {
 
     @Override
     protected void close() {
-        producer.close(30, TimeUnit.SECONDS);
-        transformationChain.close();
+        synchronized (this) {
+            tryStop();
+        }
+        if (producer != null) {
+            try {
+                producer.close(30, TimeUnit.SECONDS);
+            } catch (Throwable t) {
+                log.warn("Could not stop producer", t);
```

Comment:
Nit: the method is `Producer.close()`, not "stop". The log message should probably reflect that.